universe/System.h

```diff
@@ -73,7 +73,7 @@ class FO_COMMON_API System : public UniverseObject {
       * yet explored might be "Unexplored Region", rather than an empty string
       * for the name.  This is distinct from PublicName functions, which filter
       * the name based on ownership. */
-    [[nodiscard]] const std::string&      ApparentName(int empire_id, const Universe& u, bool blank_unexplored_and_none = false) const;
+    [[nodiscard]] const std::string       ApparentName(int empire_id, const Universe& u, bool blank_unexplored_and_none = false) const;
```

Comment:
If making it return by-value, get rid of the `const`.

Also need to check all the places `ApparentName` is called to store the result by value and not const reference.